src/MSALWrapper.Test/AuthFlow/DeviceCodeTest.cs

```diff
@@ -83,68 +84,63 @@ public async Task DeviceCodeAuthFlow_CachedToken()
         [Test]
         public async Task DeviceCodeAuthFlow_NoAccount()
         {
-            this.pcaWrapperMock.Setup(pca => pca.TryToGetCachedAccountAsync(It.IsAny<string>())).ReturnsAsync((IAccount)null);
-            this.SilentAuthUIRequired();
+            this.MockAccountNull();
             this.DeviceCodeAuthResult();
 
             // Act
             AuthFlow.DeviceCode deviceCode = this.Subject();
             var authFlowResult = await deviceCode.GetTokenAsync();
 
             // Assert
-            this.pcaWrapperMock.VerifyAll();
             authFlowResult.TokenResult.Should().Be(this.tokenResult);
             authFlowResult.TokenResult.IsSilent.Should().BeFalse();
-            authFlowResult.Errors.Should().HaveCount(1);
+            authFlowResult.Errors.Should().BeEmpty();
             authFlowResult.AuthFlowName.Should().Be("devicecode");
         }
 
         [Test]
-        public async Task DeviceCodeAuthFlow_GetTokenSilent_ReturnsNull()
+        public async Task DeviceCodeAuthFlow_GetTokenSilent_ReturnsNull_ThenSuccess()
         {
-            this.SilentAuthReturnsNull();
-
             this.MockAccount();
+            this.SilentAuthReturnsNull();
+            this.DeviceCodeAuthResult();

```

Comment:
This test is the one with the bug - where if silent auth failed by returning null, we never actually attempted auth.